openpype/hosts/houdini/plugins/publish/collect_rop_frame_range.py

```diff
@@ -2,40 +2,94 @@
 """Collector plugin for frames data on ROP instances."""
 import hou  # noqa
 import pyblish.api
+from openpype.lib import BoolDef
 from openpype.hosts.houdini.api import lib
+from openpype.pipeline import OptionalPyblishPluginMixin
 
 
-class CollectRopFrameRange(pyblish.api.InstancePlugin):
+class CollectRopFrameRange(pyblish.api.InstancePlugin,
+                           OptionalPyblishPluginMixin):
+
     """Collect all frames which would be saved from the ROP nodes"""
 
+    hosts = ["houdini"]
     order = pyblish.api.CollectorOrder
     label = "Collect RopNode Frame Range"
+    use_asset_handles = True
 
     def process(self, instance):
 
         node_path = instance.data.get("instance_node")
         if node_path is None:
             # Instance without instance node like a workfile instance
+            self.log.debug(
+                "No instance node found for instance: {}".format(instance)
+            )
             return
 
         ropnode = hou.node(node_path)
-        frame_data = lib.get_frame_data(ropnode)
+        asset_data = instance.context.data["assetEntity"]["data"]
+
+        attr_values = self.get_attr_values_from_data(instance.data)
+        if not attr_values.get("use_handles"):
+            asset_data["handleStart"] = 0
+            asset_data["handleEnd"] = 0
+
+        frame_data = lib.get_frame_data(ropnode, asset_data, self.log)
+
+        if not frame_data:
+            return
 
-        if "frameStart" in frame_data and "frameEnd" in frame_data:
+        # Log artist friendly message about the collected frame range
+        frame_start = frame_data["frameStart"]
+        frame_end = frame_data["frameEnd"]
 
-            # Log artist friendly message about the collected frame range
-            message = (
-                "Frame range {0[frameStart]} - {0[frameEnd]}"
-            ).format(frame_data)
-            if frame_data.get("step", 1.0) != 1.0:
-                message += " with step {0[step]}".format(frame_data)
-            self.log.info(message)
+        if attr_values.get("use_handles"):
+            self.log.info(
+                "Full Frame range with Handles "
+                "[{frame_start_handle} - {frame_end_handle}]\n"
+                .format(
+                    frame_start_handle=frame_data["frameStartHandle"],
+                    frame_end_handle=frame_data["frameEndHandle"]
+                )
+            )
+        else:
+            self.log.info(
+                "Use handles is deactivated for this instance, "
+                "start and end handles are set to 0.\n"
+            )
```

Comment:
```suggestion
        if attr_values.get("use_handles"):
            self.log.debug(
                "Full Frame range with Handles "
                "[{frame_start_handle} - {frame_end_handle}]"
                .format(
                    frame_start_handle=frame_data["frameStartHandle"],
                    frame_end_handle=frame_data["frameEndHandle"]
                )
            )
        else:
            self.log.debug(
                "Use handles is deactivated for this instance, "
                "start and end handles are set to 0."
            )
```
These seem like debug logs - also, no need for the new line at the end of the logs.

Reply:
I think the first one worth to be info, I also removed the redundant `"Frame range [{frame_start} - {frame_end}]"`  log
so that report looks like this

when use asset handle is on
![image](https://github.com/ynput/OpenPype/assets/20871534/5813541e-a7fa-4132-b276-13b5648be5c9)


when use asset handle is off
![image](https://github.com/ynput/OpenPype/assets/20871534/a000ac9e-5b7f-46bc-9c34-2a615da95f75)

Reply:
Interesting - I would have made **that** a debug log and instead would have logged this: https://github.com/ynput/OpenPype/pull/5698/commits/3797ad5bb7cf2cfc578534e423ac956c5994a0a5#diff-3c91d054a66e9ff7496c9c17c085ab1f33af3d8b8185a1f377c8ca57b846c09eL62-L69 but I see now that logs without handles. I think it's always worth logging the frame range that was used whether asset handles was on or off. So let's move the log to that location I linked, maybe logging as:

```python
message = "Frame range [{frame_start} - {frame_end}]"
if handle_start or handle_end:
    message += " with handles [{handle_start}]-[{handle_end}]"
self.log.info(message)
```
That way the artist facing log shows the frame range + what the handles are. The "Full frame range with handles" tells me nothing about how big the handles are.

Reply:
Fair enough!

Reply:
when use handles is off  or if handles are 0 in asset data
![image](https://github.com/ynput/OpenPype/assets/20871534/76541604-0c71-4cf4-ae13-586d51a0252e)

when use handles is on
![image](https://github.com/ynput/OpenPype/assets/20871534/86f743ee-eaf9-472f-9965-13cdb27e501e)

Reply:
I like it!